src/spikeinterface/core/sortinganalyzer.py

```diff
@@ -588,6 +588,35 @@ def load_from_zarr(cls, folder, recording=None):
 
         return sorting_analyzer
 
+    def set_temporary_recording(self, recording: BaseRecording):
+        """
+        Sets a temporary recording object. This function can be useful to temporarily set
+        a "cached" recording object that is not saved in the SortingAnalyzer object to speed up
+        computations. Upon reloading, the SortingAnalyzer object will try to reload the recording
+        from the original location in a lazy way.
+
+
+        Parameters
+        ----------
+        recording : BaseRecording
+            The recording object to set as temporary recording.
+
+        Raises
+        ------
+        ValueError
+            _description_
```

Comment:
This doesn't raise anything (technically AssertionError), but maybe we just delete this section. We often don't do the raises section.